diffpy/snmf/subroutines.py

```diff
@@ -449,76 +449,25 @@ def get_residual_matrix(component_matrix, weights_matrix, stretching_matrix, dat
-      the signals, M is the number of moments, and K is the number of components. The resulting matrix has M blocks
-      stacked horizontally where each block is the weighted and stretched components at each moment. Also contains
-      the gradient and hessian matrices of the reconstructed data.
+    2d array
+      The 2d array containing the reconstruction of input_data.
 
     """
-    stretching_factor_matrix = np.asarray(stretching_factor_matrix)
-    component_matrix = np.asarray(component_matrix)
-    weight_matrix = np.asarray(weight_matrix)
-    stretched_component_series = []
-    stretched_component_series_gra = []
-    stretched_component_series_hess = []
-    for moment in range(moment_amount):
-        for component in range(component_amount):
-            stretched_component = get_stretched_component(stretching_factor_matrix[component, moment],
-                                                          component_matrix[:, component], signal_length)
-            stretched_component_series.append(stretched_component[0])
-            stretched_component_series_gra.append(stretched_component[1])
-            stretched_component_series_hess.append(stretched_component[2])
-    stretched_component_series = np.column_stack(stretched_component_series)
-    stretched_component_series_gra = np.column_stack(stretched_component_series_gra)
-    stretched_component_series_hess = np.column_stack(stretched_component_series_hess)
-
-    reconstructed_data = []
-    reconstructed_data_gra = []
-    reconstructed_data_hess = []
-    moment = 0
-    for s_component in range(0, moment_amount * component_amount, component_amount):
-        block = stretched_component_series[:, s_component:s_component + component_amount]
-        block_gra = stretched_component_series_gra[:, s_component:s_component + component_amount]
-        block_hess = stretched_component_series_hess[:, s_component:s_component + component_amount]
-        for component in range(component_amount):
-            block[:, component] = block[:, component] * weight_matrix[component, moment]
-            block_gra[:, component] = block_gra[:, component] * weight_matrix[component, moment]
-            block_hess[:, component] = block_hess[:, component] * weight_matrix[component, moment]
-        reconstructed_data.append(block)
-        reconstructed_data_gra.append(block_gra)
-        reconstructed_data_hess.append(block_hess)
-        moment += 1
-    return np.column_stack(reconstructed_data), np.column_stack(reconstructed_data_gra), np.column_stack(
-        reconstructed_data_hess)
+    signal_length = len(components[0].iq)
+    number_of_signal = len(components[0].weights)
+    data_reconstruction = np.zeros((signal_length, number_of_signal))
+    for signal in range(number_of_signal):
+        data_reconstruction[:, signal] = reconstruct_signal(components, signal)
```

Comment:
as before, is there a reason the nd array is this way around and the code can't look like can't be:
```
data_reconstruction[signal] = reconstruct_signal(components, signal)
```
to make the code prettier?

Reply:
This function is ready to be merged. Unless you think that I should rewrite this line in the way that you suggested.